JBMC: support list-symbols

smowton opened the pull request:
Add `--list-symbols` argument to JBMC (matching existing argument in CBMC), and fix the accidental inclusion of the symbol value in the JSON output even in `list` mode.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in ODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [x] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

peterschrammel commented on jbmc/src/jbmc/jbmc_parse_options.h:
```diff
@@ -59,7 +59,7 @@ class optionst;
   OPT_SHOW_CLASS_HIERARCHY \
   "(show-loops)" \
-  "(show-symbol-table)(show-parse-tree)" \
+  "(show-symbol-table)(list-symbols)(show-parse-tree)" \
```
Preferably put on separate lines.

peterschrammel reviewed (approved):
Clang-format is complaining: https://travis-ci.org/diffblue/cbmc/jobs/443779256#L547

tautschnig reviewed (changes requested):
I am struggling to reconcile the commit subject with the actual changes. The notes in the PR make more sense, but seemingly those haven't made it into the commit message.

smowton committed:
list-symbols + json-ui: don't print symbol value

The plain-text mode correctly prints only names and types; with --json-ui until now
we accidentally printed the symbol's value as well, which should only happen with --show-symbol-table.

smowton committed:
Fix style of show-symbol-tables call

No functional changes

smowton committed:
JBMC: add list-symbols option

This mirrors the support already present in cbmc

smowton commented:
@tautschnig oops, was from a subtree cherry-pick. Fixed.

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 74c9ff9351337a571ad708ad1d22d574f547c688).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88760621

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 87581eb741ef24ce4a9259adae65404189568fa7).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88799241
